src/Highlight.ts

```diff
@@ -91,17 +101,48 @@ export default class Highlight {
   }
 
   public serialize(referenceElement?: HTMLElement): SerializedHighlight {
-    referenceElement = referenceElement || dom(this.range.commonAncestorContainer).closest('[id]');
+    const validReferenceElement = this.getValidReferenceElement(referenceElement);
+
+    if (!validReferenceElement) {
+      throw new Error('reference element not found');
+    }
 
     return new SerializedHighlight({
       ...this.data,
-      ...SerializedHighlight.defaultSerializer(this.range, referenceElement),
+      ...SerializedHighlight.defaultSerializer(this.range, validReferenceElement),
     });
   }
   private loadStyle() {
-    const {style} = this.data;
+    const { style } = this.data;
     if (style) {
       this.elements.forEach((element) => element.classList.add(style));
     }
   }
+
+  private checkReferenceElement(referenceElement?: HTMLElement): boolean {
```

Comment:
this conversation is not resolved https://github.com/openstax/highlighter/pull/31#discussion_r357793499